### PR TITLE
chore(config): entity type registration no longer uses global $CONFIG, adds tests

### PR DESCRIPTION
**Pull request by hypeJunction**

### CODE REVIEW

File: engine/lib/entities.php [hunk 1]
```diff
@@ -642,8 +639,13 @@ function is_registered_entity_type($type, $subtype = null) {
...
  * @return string A viewable list of entities
  * @since 1.7.0
+ * @deprecated 2.3 Use elgg_list_entities() with 'type_subtype_pairs' options from {@link get_registered_entity_types}
  */
 function elgg_list_registered_entities(array $options = array()) {
+
+	elgg_deprecated_notice(__FUNCTION__ . ' has been deprecated. '
+			. 'Use elgg_list_entities() with "type_subtype_pairs" options from get_registered_entity_types()', '2.3');
+
 	elgg_register_rss_link();
```
Comment (mrclay) [inline engine/lib/entities.php]: If devs rewrite using elgg_list_entities(), will they need to call this?

**Comment by mrclay** [comment]
What's the motivation for retiring elgg_list_registered_entities()? Is it standing in the way of something?

**Comment by hypeJunction** [comment]
I couldn't think of a use case that would justify keeping all that untestable code.

**Comment by mrclay** [comment]
It's been around forever and the benefit of removing it is almost nil.

**Comment by hypeJunction** [comment]
Fine by me.
I think I am going to take a leave of absense from Elgg and enjoy the
summer. Noone seems to have time to review any of the pull requests, and I
don't want to be the idiot that missed the summer chasing a higher purpose.
So, I am going to abandon this and other pull requests until after
mid-September.

**Comment by hypeJunction** [comment]
Replaced by #10069

**Pull request closed without merging**